### PR TITLE
client code for XQueueJob

### PR DESCRIPTION
**Pull request by scostache**
Added listener and informer for XQueueJobs, as well as a dummy controller

### CODE REVIEW

**Comment by k82cn** [comment]
please add test for this, e2e or unit test.

**Comment by scostache** [comment]
I have followed the example in test/e2e.go and added a test for XQueueJob

**Comment by scostache** [comment]
@k82cn : e2e test is added for this branch. is it something else required to be merged?

File: pkg/controller/queuejob/queuejob_controller_ex.go [hunk 1]
```diff
@@ -0,0 +1,240 @@
+/*
```
Comment (k82cn) [inline pkg/controller/queuejob/queuejob_controller_ex.go]: please add this controller to his own directory.

File: pkg/controller/queuejob/queuejob_controller_ex.go [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+
+	// ControllerUIDLabel label string for queuejob controller uid
+	ControllerUIDLabel string = "controller-uid"
+)
+
+// controllerKind contains the schema.GroupVersionKind for this controller type.
+var controllerKind = arbv1.SchemeGroupVersion.WithKind("XQueueJob")
+
+//XController the XQueueJob Controller type
+type XController struct {
```
Comment (k82cn) [inline pkg/controller/queuejob/queuejob_controller_ex.go]: My understand is that : this PR is only for client, so why XQueueJob controller here?

Comment (scostache) [reply pkg/controller/queuejob/queuejob_controller_ex.go]: It is a dummy controller that only listens for events

**Comment by scostache** [comment]
@k82cn : I separated the controller for XQueueJob in a new directory

**Pull request closed without merging**